### PR TITLE
TCOTA-554 TCOTA-556 fixing assignment move issues

### PR DESCRIPTION
**Pull request by relong76**
- added the treatment copy functionality
- fixed JPA database cascading issue for MC question join table

### CODE REVIEW

File: src/main/java/edu/iu/terracotta/repository/AssignmentRepository.java [hunk 1]
```diff
@@ -6,11 +6,13 @@
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Query;
-import org.springframework.transaction.annotation.Transactional;
 
 import java.util.List;
 
+import javax.transaction.Transactional;
```
Comment (machristie) [inline src/main/java/edu/iu/terracotta/repository/AssignmentRepository.java:12]: I'm curious why the switch to the javax Transactional annotation.  I'm not really aware of the pros or cons but I found this (https://stackoverflow.com/questions/26387399/javax-transaction-transactional-vs-org-springframework-transaction-annotation-tr) and one of the answers indicates that you shouldn't mix and match javax Transactional with spring's Transactional annotation, so I just want to make sure this won't cause any unwanted issues down the road.

File: src/main/resources/db/changelog/2022.10/06-01-changelog.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:pro="http://www.liquibase.org/xml/ns/pro" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/pro http://www.liquibase.org/xml/ns/pro/liquibase-pro-latest.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">
+    <changeSet author="rlong (generated)" id="1665057691865-1">
+        <dropForeignKeyConstraint baseTableName="terr_question_mc" constraintName="FK_TERR_QUESTION_MC_ON_QUESTION_ID"/>
+    </changeSet>
+    <changeSet author="rlong (generated)" id="1665057691865-2">
```
Comment (machristie) [inline src/main/resources/db/changelog/2022.10/06-01-changelog.xml:6]: I like this, but with this in place is the change in deleteById still necessary? Seems like the questionMcRepository.deleteByQuestionId is no longer needed.

**Review by machristie — COMMENTED** [review]
Hi @relong76 , this looks pretty good to me but I just have a couple questions. See the comments.

File: src/main/java/edu/iu/terracotta/repository/AssignmentRepository.java (hunk 1 above)
Comment (relong76) [reply src/main/java/edu/iu/terracotta/repository/AssignmentRepository.java:12]: oops! That was an IDE syntax completion selection mistake. I'll fix!

File: src/main/resources/db/changelog/2022.10/06-01-changelog.xml (hunk 2 above)
Comment (relong76) [reply src/main/resources/db/changelog/2022.10/06-01-changelog.xml:6]: Good catch! That was from when i was diagnosis the issue with foreign key constraint issues on deletes of MC questions. I will fix it.

**Comment by relong76** [comment]
@machristie I've made the requested changes. Should be good to go!

File: src/main/java/edu/iu/terracotta/repository/QuestionRepository.java [hunk 3]
```diff
@@ -2,11 +2,12 @@
 
 import edu.iu.terracotta.model.app.Question;
 import org.springframework.data.jpa.repository.JpaRepository;
-import org.springframework.transaction.annotation.Transactional;
 
 import java.util.List;
 import java.util.Optional;
 
+import javax.transaction.Transactional;
```
Comment (machristie) [inline src/main/java/edu/iu/terracotta/repository/QuestionRepository.java:9]: I forgot to mention this one, I'll fix it

**Review by machristie — APPROVED** [review]
Looks good 👍

File: src/main/java/edu/iu/terracotta/repository/QuestionRepository.java (hunk 3 above)
Comment (relong76) [reply src/main/java/edu/iu/terracotta/repository/QuestionRepository.java:9]: Doh! Thanks! :)

Comment (relong76) [reply src/main/java/edu/iu/terracotta/repository/QuestionRepository.java:9]: I blame VS Code.... I just switched a few weeks ago from Eclipse and I'm getting used to the order of syntax suggestions it provides... Eclipse always did the Spring deps 1st in Spring Boot apps

**Pull request merged**